net/seafile-server/Makefile

```diff
@@ -40,7 +40,7 @@ define Package/seafile-server
     TITLE:=Seafile server
     URL:=https://seafile.com/
     DEPENDS:=+libarchive +libopenssl +glib2 +libsearpc +seafile-ccnet +seafile-seahub +sqlite3-cli +python-mysql +python-urllib3 \
```

Comment:
Number of DEPENDS is huge. On first glance, libopenssl can be removed as it is implied by +libevent2-openssl. I assume that is the case with several others.

Reply:
My opinion on this is that a package should not count on another dependent to pull up its own dependencies.  So, if this has a direct usage of openssl, which it does, then libopenssl should be here.  I imagine it will hardly be the case of libevent2-openssl, but packages may change crypto providers.  It would be a good exercise to test all of these dependencies, but it's beyond the scope of what I'm doing here.  I vote for keeping at least libopenssl.  What do you think?

Reply:
There's prior art that goes against this. For example, https://github.com/openwrt/packages/blob/master/libs/ibrdtn/Makefile#L28

This is a C++ package, so you would expect a dependency on libstdcpp but that gets pulled in by ibrcommon.

My view is that these dependencies should be kept simple.

Reply:
It's not a matter of simplicity, but of correctness/robustness.  If ibrcommon were to change to straight C during an upgrade (granted, it mostly happens the other way), then we would have to go searching for breakage that can be otherwise avoided.
Just because there are packages doing this, does not mean we should be blindly following it.  If this was discussed before, It would be another case.
If you insist again :smile:, then I'll change it.

Reply:
Well, let's let @commodo decide.

Notice that in https://github.com/openwrt/packages/pull/9035 , I add ```include $(INCLUDE_DIR)/uclibc++.mk``` to packages that have no libstdcpp dependency but imply it.

Reply:
My keyboard is malfuncioning.  `sed "s/change it/kill you"` :rofl:.  @commodo is a good moderator.

Reply:
@neheb @cotequeiroz 
to put things in perspective a bit: i had some hesitance about taking ownership of this package from Gergely Kiss ;
before i did that, I saw how this (package-file) was implemented, and I slightly face-palmed;
then i slightly panic-ed, and decided not to take it;
then i did;

the idea is that (when i did take it) i admitted to myself that this will need a lot of cleanup, and breakage (when doing cleanups);
so, yeah: this (the number of depends) is bad (maybe even pretty bad) ; and, even seafile itself is pretty bad ;
@cotequeiroz  already fixed some things i broke about this; (btw: i do promise to try better with updating this), and I thank you for this;

the only thing i can say, is: let's go ahead with this as-is here, and we will have to try to fix this better (the dependency list, the makefile build rules, etc); and let's fix this incrementally, as there will be plenty of breakage (whether we want to or not).

thanks
Alex